tools/php-cs-fixer/composer.json

```diff
@@ -1,5 +1,6 @@
 {
     "require": {
+        "php": "8.3",
```

Comment:
This is very high, and could block some contribution IMHO. WDYT?

Reply:
It's about ensuring that the checks run in the same environment locally as in CI. I'm not certain if running php-cs-fixer with PHP 8.1 versus 8.3 will lead to discrepancies, but with phpstan, it's definitely possible and I experienced it on my own lib.

To avoid situations where everything appears fine locally but then CI fails, causing frustration and unnecessary feedback loops, I'm making sure the local and CI environments are identical by more or less pinning the php version.

Reply:
Does this situation exist?
Never heard about this in castor.

Reply:
I can give you an example from my own lib, as part of [PHP plan to refactor resource to object](https://php.watch/articles/resource-object) some [pspell functions signatures were changed](https://php.watch/versions/8.1/PSpell-resource#PSpell-Dictionary) and thus running PHPStan with 8.0 or 8.1 would produce different results as the returned type is different.

[Tests are green because you do a `PHP_VERSION_ID` check in your code](https://github.com/tigitz/php-spellchecker/blob/master/src/Spellchecker/PHPPspell.php#L65) but PHPStan cannot do that.

To solve this you could either choose arbitrarily that Static Anlysis check must be done in the most up-to-date PHP versions and force contributors to use this version (what I'm suggesting here for simplicity) or [manage different phpstan config for each problematic php versions](https://github.com/tigitz/php-spellchecker/blob/master/phpstan.8.2.neon).

The idea is to avoid situations where users will be like "hey it fails on my local" and you'll wonder why because CI is green. So forcing the php running version of the tools will tell the user right away "you're not in the required environment to proceed with the check" and solve the issue at its root.

As Castor seems to focus on using php primitives as much as possible in its helpers, I believe it's a risk that's worth preventing. IMO contributors are more likely to be knowledgeable enough to be able to run different version of PHP.

Open to remove this if it proves unconvincing.

Reply:
> But we don't support php 8

Yeah my point was more about that PHP can break its method signature in the future as it happened in php 8.

> The day such situations will occurs, we'll adapt.
>
> Sorry

Sure that's fine too, just tried to share my own experience :slightly_smiling_face: